agent/rpc/peering/service.go

```diff
@@ -427,6 +431,19 @@ type BidirectionalStream interface {
 
 // StreamResources handles incoming streaming connections.
 func (s *Service) StreamResources(stream pbpeering.PeeringService_StreamResourcesServer) error {
+	if s.Backend == nil {
+		s.logger.Error("cannot establish stream without a backend", "error", errPeeringBackendNil)
+		return errPeeringBackendNil
+	}
```

Comment:
I don't think this is necessary as no other endpoints check this. You will get a nil pointer dereference elsewhere anyway, and a failure to have configured a backend is a programmer error, not something that will happen at runtime.

Reply:
taken out